modules/ui/init.js

```diff
@@ -299,7 +298,7 @@ export function uiInit(context) {
 
         var osm = context.connection();
         if (!_initCounter++) {
-            if (!hash.startWalkthrough) {
+            if (!ui.hash.startWalkthrough) {
                 if (context.history().lock() && context.history().restorableChanges()) {
```

Comment:
When I built/ran this version of rapid locally, it worked fine for edits but then if I refreshed the page, I did not get asked by rapid if I wanted to restore my edits from a previous session. It seems that something in the cookie/locking or context.storage mechanisms got messed up. 

This bubbles up to modules/ui/init.js' check to see if there are restorable changes: 
![image](https://user-images.githubusercontent.com/1887955/65181503-3212ff00-da2d-11e9-80b0-0123de93125c.png)

the check for restorableChanges fails because `context.storage(getKey('saved_history'))` is always `null`. 

This led me to look at the window.localStorage in the devTools. 'Good' version of current RapiD is on the left- it has the iD_https://mapwith.ai_saved_history, which is what this 'getKey('saved_history')' should return. 

Note that on the right (local copy running this branch) there is no such saved_history info! # # @ @

Reply:
Did you happen to have multiple tabs or windows open for iD or RapiD? I noticed that the restore window wouldn't show up if another iD/RapiD window is already open.

Reply:
I did indeed- I was testing other changes locally and comparing them with what we have in prod when I switched to this task. 

Hypothesis: multiple rapid windows might be okay as long as they're all hosted on the same domain. As soon as you have multiple copies running cross domain, it seems possible that the cookie tracking (which is prepended with the site domain) breaks down. Let's see...

Reply:
Never mind! After closing chrome and re-opening, then rebuilding the app from a cleaned git directory (`git clean -dfx`, `npm install`, `npm run all`) things cleared up. Sorry for the false alarm!

Reply:
iD won't offer to restore history when multiple windows are open, that's what the "lock" in `modules/core/history.js` is for: https://github.com/openstreetmap/iD/blob/87d0fc0a16d391091a464402e752c01909a89c51/modules/core/history.js#L20